File: multiple-languages/go/alicloudroscdkaligreen/IBizType.go
package alicloudroscdkaligreen

import (
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"

	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkaligreen/internal"
	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkcore"
)

// Represents a `BizType`.
type IBizType interface {
	alicloudroscdkcore.IResource
	// Attribute BizTypeName: The name of the business scenario defined by the customer.
	AttrBizTypeName() interface{}
	// Attribute CiteTemplate: Specifies whether to import the configuration of an industry template.
	//
	// Default value: false.
	AttrCiteTemplate() interface{}
	// Attribute Description: The description of the business scenario defined by the customer.
	AttrDescription() interface{}
	// Attribute IndustryInfo: The industry classification.
	AttrIndustryInfo() interface{}
	Props() *BizTypeProps
}

// The jsii proxy for IBizType
type jsiiProxy_IBizType struct {
	internal.Type__alicloudroscdkcoreIResource
}

func (j *jsiiProxy_IBizType) AttrBizTypeName() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrBizTypeName",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IBizType) AttrCiteTemplate() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrCiteTemplate",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IBizType) AttrDescription() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrDescription",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IBizType) AttrIndustryInfo() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrIndustryInfo",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IBizType) Props() *BizTypeProps {
	var returns *BizTypeProps
	_jsii_.Get(
		j,
		"props",
		&returns,
	)
	return returns
}

File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Aligreen/AlibabaCloud/SDK/ROS/CDK/Aligreen/IBizType.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Aligreen
{
    /// <summary>Represents a `BizType`.</summary>
    [JsiiInterface(nativeType: typeof(IBizType), fullyQualifiedName: "@alicloud/ros-cdk-aligreen.IBizType")]
    public interface IBizType : AlibabaCloud.SDK.ROS.CDK.Core.IResource
    {
        /// <summary>Attribute BizTypeName: The name of the business scenario defined by the customer.</summary>
        [JsiiProperty(name: "attrBizTypeName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrBizTypeName
        {
            get;
        }

        /// <summary>Attribute CiteTemplate: Specifies whether to import the configuration of an industry template.</summary>
        /// <remarks>
        /// Default value: false.
        /// </remarks>
        [JsiiProperty(name: "attrCiteTemplate", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrCiteTemplate
        {
            get;
        }

        /// <summary>Attribute Description: The description of the business scenario defined by the customer.</summary>
        [JsiiProperty(name: "attrDescription", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrDescription
        {
            get;
        }

        /// <summary>Attribute IndustryInfo: The industry classification.</summary>
        [JsiiProperty(name: "attrIndustryInfo", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrIndustryInfo
        {
            get;
        }

        [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-aligreen.BizTypeProps\"}")]
        AlibabaCloud.SDK.ROS.CDK.Aligreen.IBizTypeProps Props
        {
            get;
        }

        /// <summary>Represents a `BizType`.</summary>
        [JsiiTypeProxy(nativeType: typeof(IBizType), fullyQualifiedName: "@alicloud/ros-cdk-aligreen.IBizType")]
        internal sealed class _Proxy : DeputyBase, AlibabaCloud.SDK.ROS.CDK.Aligreen.IBizType
        {
            private _Proxy(ByRefValue reference): base(reference)
            {
            }

            /// <summary>Attribute BizTypeName: The name of the business scenario defined by the customer.</summary>
            [JsiiProperty(name: "attrBizTypeName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrBizTypeName
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute CiteTemplate: Specifies whether to import the configuration of an industry template.</summary>
            /// <remarks>
            /// Default value: false.
            /// </remarks>
            [JsiiProperty(name: "attrCiteTemplate", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrCiteTemplate
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute Description: The description of the business scenario defined by the customer.</summary>
            [JsiiProperty(name: "attrDescription", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrDescription
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute IndustryInfo: The industry classification.</summary>
            [JsiiProperty(name: "attrIndustryInfo", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrIndustryInfo
            {
                get => GetInstanceProperty<object>()!;
            }

            [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-aligreen.BizTypeProps\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Aligreen.IBizTypeProps Props
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Aligreen.IBizTypeProps>()!;
            }

            /// <summary>The environment this resource belongs to.</summary>
            /// <remarks>
            /// For resources that are created and managed by the CDK
            /// (generally, those created by creating new class instances like Role, Bucket, etc.),
            /// this is always the same as the environment of the stack they belong to;
            /// however, for imported resources
            /// (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
            /// that might be different than the stack they were imported into.
            /// </remarks>
            [JsiiProperty(name: "env", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ResourceEnvironment\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment Env
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment>()!;
            }

            /// <summary>The stack in which this resource is defined.</summary>
            [JsiiProperty(name: "stack", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.Stack\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.Stack Stack
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.Stack>()!;
            }

            /// <summary>The construct tree node for this construct.</summary>
            [JsiiProperty(name: "node", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ConstructNode\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode Node
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode>()!;
            }
        }
    }
}
